### PR TITLE
Document RPC framework

### LINKED ISSUE #500: Document NextGen RPC subsystem
We added some framework code to support better RPC handling testability and separation of concerns in #494 .
Some documentation on how to use it would be a good addition.

### PR DESCRIPTION
**Pull request by godexsoft**
Fixes #500

Adding some Readme and more detailed doxygen comments for public member functions.

### CODE REVIEW

File: src/rpc/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+See `common` subfolder.
+
+- **AnyHandler**: The type-erased wrapper that allows for storing different handlers in one map/vector.
+- **RpcSpec/FieldSpec**: The RPC specification classes, used to specify how incoming JSON is to be validated before it's parsed and passed on to individual handler implementations.
+- **Validators**: A bunch of supported validators that can be specified as requirements for each **`FieldSpec`** to make up the final **`RpcSpec`** of any given RPC handler.
+
+## Implementing a (NextGen) handler
+See `unittests/rpc` for exmaples.
+
+Handlers need to fullfil the requirements specified by the **`Handler`** concept (see `rpc/common/Concepts.h`):
```
Comment (cindyyan317) [inline src/rpc/README.md]: My English is limited. Is fullfil a word?

Comment (godexsoft) [reply src/rpc/README.md]: No, it meant to say fulfil (or fulfill) 👍

**Review by cindyyan317 — CHANGES_REQUESTED** [review]
send back just to check 'fullfil' is a word or not?

**Review by cindyyan317 — APPROVED** [review]
Screenshot 2023-02-03 at 09 10 43

**Comment by godexsoft** [comment]
> send back just to check 'fullfil' is a word or not?

correct spelling is `fulfil` or `fulfill`. fixed 👍

**Pull request merged**